avocado/utils/pmem.py

```diff
@@ -0,0 +1,298 @@
+                '%s list %s' % (self.ndctl, option), shell=True))
+        except ValueError:
+            json_op = []
+        return json_op
+
+    @staticmethod
+    def run_ndctl_list_val(json_op, field):
+        """
+        Get the value of a field in given json
+
+        :param json_op: Input Json object
+        :param field: Field to find the value from json_op object
+
+        :rtype: Found value type, None ig not found
+        """
+        for key, value in json_op.items():
+            if key == field:
+                return value
+        return None
+
+    def run_daxctl_list(self, options=''):
+        """
+        Get the json of each provided options
+
+        :param options: optional arguments to daxctl list command
+        :return: By default returns entire list of json objects
+        :rtype: list of json objects
+        """
+        return json.loads(process.system_output(
+            '%s list %s' % (self.daxctl, options), shell=True))
+
+    def get_slot_count(self, region):
+        """
+        Get max slot count in the index area for a  dimm backing a region
+        We use region0 - > nmem0
+
+        :param region: Region for which slot count is found
+        :return: Number of slots for given region
+        :rtype: int
+        """
+        nmem = "nmem%s" % re.findall(r'\d+', region)[0]
+        try:
+            json_op = json.loads(process.system_output(
+                '%s read-labels -j %s ' % (self.ndctl, nmem), shell=True))
+        except ValueError:
+            json_op = []
+        first_dict = json_op[0]
+        index_dict = self.run_ndctl_list_val(first_dict, 'index')[0]
+        return self.run_ndctl_list_val(index_dict, 'nslot') - 2
+
+    @staticmethod
+    def is_region_legacy(region):
+        """
+        Check whether we have label index namespace. If legacy we can't create
+        new namespaces.
+
+        :param region: Region for which legacy check is made
+        :return: True if given region is legacy, else False
+        """
+        nstype = genio.read_file("/sys/bus/nd/devices/%s"
+                                 "/nstype" % region).rstrip("\n")
+        if nstype == "4":
+            return True
+        return False
+
+    @staticmethod
+    def check_buses():
+        """
+        Get buses from sys subsystem to verify persisment devices exist
+
+        :return: List of buses available
+        :rtype: list
+        """
+        return glob.glob('/sys/bus/nd/drivers/nd_bus/ndbus*')
+
+    def disable_region(self, name='all'):
+        """
+        Disable given region
+
+        :param name: name of the region to be disabled
+        :return: True on success
+        :raise: :class:`PMemException`, if command fails.
+        """
+        if process.system('%s disable-region %s' % (self.ndctl, name),
+                          shell=True, ignore_status=True):
+            raise PMemException("Failed to disable %s region(s)" % name)
```

Comment:
raising `Exception` is fine but library either return some value like in this case True/False as a success it returns true so I feel in Fail condition it should return False as it will easy to process/use this library in the Test program

Reply:
Returning failure can also be used. But the main idea behind raising an user-defined Exception is to easily handle it from the test case. For example if a test uses 4 to 5 functions of the library, instead of a `if` check for each of the function/cmd, isn't it better to handle the exception from the test? And if the functions are used across different tests then these have to be repeated everywhere. Because these libraries are expected to work most often than not.

Reply:
@harish-24 yes i got your idea earlier as well , but library should return True and False that is easy to handel

Reply:
@PraveenPenguin It could be any of the two solutions. It depends on the library usage. This case is very similar to `process.run` which does something similar. What is your thought on this?

Reply:
yes but in that case, you need to always use try-expect in test to Handel somehow I did not like that as library always returns some value and in your case, you are returning `True`  in success case , In process API case i relate like a wrapper over python library (os.system) so it seems me ok

Reply:
anyway this is just a suggestion you take your call

Reply:
Sure, I would like to go with existing approach as it is more convenient from handling in a test and same is included in docstring.